### ISSUE TITLE
Upcoming major v4 release

### ISSUE THREAD

**Issue by scottsfarley93** [mapbox/mapbox-gl-geocoder#197]
There are several outstanding issues and feature improvements that would improve the usability, look/feel, and result relevance of the plugin. I suspect that some of these changes may be large and/or breaking, so I think targeting this as a v4 major release seems reasonable. 

Here's the list of tickets that we've prioritized but totally open to suggestions about other features/bugs that should also make the list. 

mapbox/mapbox-gl-geocoder#51 - camera animation properties
mapbox/mapbox-gl-geocoder#99 - result event emitted twice
mapbox/mapbox-gl-geocoder#115 - expose the clear method as public
mapbox/mapbox-gl-geocoder#162 -  accessibility issues
mapbox/mapbox-gl-geocoder#153 -  include matching text
mapbox/mapbox-gl-geocoder#176 - expose setOptions method
mapbox/mapbox-gl-geocoder#150 -  localize placeholder text 
mapbox/mapbox-gl-geocoder#89 - mobile styling
mapbox/mapbox-gl-geocoder#167 - minor style tweak
https://github.com/mapbox/mapbox-gl-geocoder/issues/157 - async local geocoder
#195 - new default options
#196 - feedback widget

\cc @mapbox/search @katydecorah @yuletide @kbauhaus @andrewharvey

**Comment by scottsfarley93** [comment]
Noting several other features/fixes that will be included:
- #219 - show a mapbox-gl marker on the map
- #216 - update dev dependencies to avoid security vulnerabilities
- #217 - remove checks for unstable feature ids

**Comment by andrewharvey** [comment]
To help track this, should we close issues when they are merged into the version4 branch?

**Comment by scottsfarley93** [comment]
@andrewharvey we can certainly do that. It might be confusing, however, since the changes will not be deployed in the plugin until version 4 lands in master -- they'll just be waiting in the version 4 working branch. Happy to do whatever makes the most sense.

**Comment by andrewharvey** [comment]
GL JS closes tickets once the PR is merged, not when the release is made. I think that's a good model, it makes it clearer which issues still need to be implemented and which are already done.

My vote would be we close them once merged into v4 branch.

**Comment by scottsfarley93** [comment]
👍 we can start doing this.

**Comment by scottsfarley93** [comment]
The version 4 working branch is looking pretty close to being in shape for an initial beta release. I think we can ship a `v4.0.0-beta.0` at some point tomorrow after which time we can ensure all the [examples](https://github.com/mapbox/mapbox-gl-js/pull/8053) work as expected. We can address any problems we uncover in the beta over the next week or two and then plan to cut a new full release around April 8.

**Comment by scottsfarley93** [comment]
🎉 v4.0.0 released!

See https://github.com/mapbox/mapbox-gl-geocoder/releases/tag/v4.0.0 and https://www.npmjs.com/package/@mapbox/mapbox-gl-geocoder/v/4.0.0.
